sql/core/src/test/scala/org/apache/spark/sql/streaming/FileStreamSinkSuite.scala

```diff
@@ -575,6 +576,43 @@ abstract class FileStreamSinkSuite extends StreamTest {
       }
     }
   }
+
+  test("formatCheck flag") {
+    withSQLConf(
+      "fs.file.impl" -> classOf[FailFormatCheckFileSystem].getName,
```

Comment:
not needed here, but it's worth knowing that hadoop-common-test jar has a class `org.apache.hadoop.fs.FileSystemTestHelper` that lets you inject an instance of an FS into the cache. Very useful from time to time. Spark could either use that  or add its own (test module) class to o.a.h.fs and call `FileSystem.addFileSystemForTesting()` when needed

Reply:
Thanks for the guidance, let me check the usage of FileSystemTestHelper.